lab-nassir/server.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+//node modules
+const http = require('http');
+const querystring = require('querystring');
+const url = require('url');
+//npm modules
+const cowsay = require('cowsay');
+//app modules
+const bodyParser = require('./lib/body-parser');
+//env variables
+const PORT = process.env.PORT || 3000;
+//module constants
+//module logic
+
+const server = http.createServer(function(req, res){
+  req.url = url.parse(req.url);
+  req.url.query = querystring.parse(req.url.query);
+  console.log('query', req.url.query);
+
+  if (req.url.pathname === '/') {
+    res.writeHead(200, {'Content-Type': 'text/plain'});
+    res.write('Hello World!');
+    res.end();
+  }
+
+  if(req.method === 'GET' && req.url.pathname === '/cowsay') {
+    if (!req.url.query.text) {
+      res.writeHead(400, {'Content-Type': 'text/plain'});
+      res.write(cowsay.say({text: 'bad request\ntry: http localhost:3000/cowsay text==howdy', f: req.url.query.f}));
+      res.end();
+    } else if (req.url.query.text) {
+      res.writeHead(200, {'Content-Type': 'text/plain'});
+      res.write(cowsay.say({text: req.url.query.text, f: req.url.query.f}));
+      res.end();
+    }
+  }
+
+  if(req.method === 'POST' && req.url.pathname === '/cowsay') {
+    bodyParser(req, function(err){
+      console.log('bodyParser req body: ', req.body);
+      if (err) {
+        res.writeHead(400, {'Content-Type': 'text/plain'});
+        res.write(cowsay.say({text: 'bad request\ntry: including a JSON file with {text: message}', f: req.url.query.f}));
+        console.error('ERROR ERROR: bodyParser callbacking an error');
+        res.end();
+      } else if (!req.body.text) {
+        res.writeHead(400, {'Content-Type': 'text/plain'});
+        res.write(cowsay.say({text: 'bad request\ntry: including a "text" parameter in your JSON file', f: req.url.query.f}));
+        res.end();
+      } else if (req.body.text) {
```

Comment:
You could also return out of your above conditions and then this doesn't need to be in an else if conditional, and is just the default behavior.